pkg/publicapi/endpoint/requester/endpoints_result.go

```diff
@@ -25,32 +25,29 @@ import (
 //	@Failure				400				{object}	string
 //	@Failure				500				{object}	string
 //	@Router					/api/v1/requester/results [post]
-func (s *Endpoint) results(res http.ResponseWriter, req *http.Request) {
-	ctx := req.Context()
+func (s *Endpoint) results(c echo.Context) error {
 	var stateReq apimodels.StateRequest
-	if err := render.DecodeJSON(req.Body, &stateReq); err != nil {
-		http.Error(res, err.Error(), http.StatusBadRequest)
-		return
+	if err := c.Bind(&stateReq); err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
 	}
-	res.Header().Set(apimodels.HTTPHeaderClientID, stateReq.ClientID)
-	res.Header().Set(apimodels.HTTPHeaderJobID, stateReq.JobID)
+	c.Response().Header().Set(apimodels.HTTPHeaderClientID, stateReq.ClientID)
+	c.Response().Header().Set(apimodels.HTTPHeaderJobID, stateReq.JobID)
 
+	ctx := c.Request().Context()
 	ctx = system.AddJobIDToBaggage(ctx, stateReq.JobID)
 	system.AddJobIDFromBaggageToSpan(ctx, oteltrace.SpanFromContext(ctx))
 
 	executions, err := s.jobStore.GetExecutions(ctx, stateReq.JobID)
 	if err != nil {
-		http.Error(res, err.Error(), http.StatusInternalServerError)
-		return
+		return c.String(http.StatusInternalServerError, err.Error())
 	}
 
 	results := make([]model.PublishedResult, 0)
 	for _, execution := range executions {
 		if execution.ComputeState.StateType == models.ExecutionStateCompleted {
 			storageConfig, err := legacy.ToLegacyStorageSpec(execution.PublishedResult)
 			if err != nil {
-				http.Error(res, err.Error(), http.StatusInternalServerError)
-				return
+				return c.String(http.StatusInternalServerError, err.Error())
```

Comment:
Why are we returning strings instead of json?

Reply:
good catch